packages/react-i18n/src/babel-plugin/index.ts

```diff
@@ -114,45 +98,72 @@ export default function injectWithI18nArguments({
         if (nodePath.node.source.value !== '@shopify/react-i18n') {
           return;
         }
+
         const {specifiers} = nodePath.node;
-        for (const specifier of specifiers) {
+        specifiers.forEach((specifier, index) => {
           if (
             !t.isImportSpecifier(specifier) ||
-            (specifier.imported.name !== 'withI18n' &&
-              specifier.imported.name !== 'useI18n')
+            !I18N_CALL_NAMES.includes(specifier.imported.name)
           ) {
-            continue;
+            return;
           }
 
           const bindingName = specifier.local.name;
           const binding = nodePath.scope.getBinding(bindingName);
 
-          if (binding != null) {
-            const fallbackID = nodePath.scope.generateUidIdentifier('en').name;
-            const {filename} = this.file.opts;
-
-            addI18nArguments({
-              binding,
-              bindingName,
-              filename,
-              fallbackID,
-              insertImport() {
-                const {program} = state;
+          if (!binding) {
+            return;
+          }
+
+          const {mode} = state.opts;
+          const fromGeneratedIndex = mode === 'from-generated-index' || false;
+          const fallbackLocale = DEFAULT_FALLBACK_LOCALE;
+
+          const fallbackID = nodePath.scope.generateUidIdentifier(
+            camelCase(fallbackLocale),
+          ).name;
+          const translationArrayID = fromGeneratedIndex
+            ? `__shopify__i18n_translations_${index}`
```

Comment:
I think only one `index.js` file can exist, so I'm not sure why `index` is needed?

There's [a check to bail when someone uses `useI18n` / `withI18n` multiple times](https://github.com/Shopify/quilt/blob/01cae2989c7bc430953e8a26bbee1ef20d64e2b6/packages/react-i18n/src/babel-plugin.ts#L79-L84), so I _think_ we can only hit this path once.

Reply:
:+1: will remove